Handling concatenating lists with `#` property

phillipskevin opened the pull request:
Closes https://github.com/canjs/can-define/issues/95.

phillipskevin commented:
@justinbmeyer does this look like the correct way to handle this?

justinbmeyer commented:
DefineMap as a static property is gone and shouldn't be used. I'm. It sure if this is the perfect fix, how are types converted elsewhere? 

Sent from my iPhone

> On Dec 5, 2016, at 12:14 PM, Kevin Phillips <notifications@github.com> wrote:
> 
> @justinbmeyer does this look like the correct way to handle this?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

phillipskevin commented:
Other places are using `this.__type(...)` to do conversion. `this.__type` internally checks if the new value is an instanceof the `['#'].Type`. I've updated this PR to use this approach and removed the `serializeNonTypes` function that is no longer needed.
